src/Microsoft.DotNet.Build.Tasks/Targets/resources.targets

```diff
@@ -1,59 +1,38 @@
+    <ResourceSRPath Condition="'$(ResourceSRPath)'==''">$(ResourcesSourceOutputDirectory)\SR.cs</ResourceSRPath>
+    <IntermediateResOutputFileFullPath Condition="'$(MSBuildProjectExtension)' == '.csproj'">$(IntermediateOutputPath)SR.cs</IntermediateResOutputFileFullPath>
+    <IntermediateResOutputFileFullPath Condition="'$(MSBuildProjectExtension)' == '.vbproj'">$(IntermediateOutputPath)SR.vb</IntermediateResOutputFileFullPath>
+  </PropertyGroup>
+  
+  <PropertyGroup Condition="'$(StringResourcesPath)'!=''">
+      <CompileDependsOn>
+          GenerateResourcesSource;
+          $(CompileDependsOn);
+      </CompileDependsOn>
   </PropertyGroup>
 
   <Target Name="GenerateResourcesSource"
-      Condition="'$(ResInputPath)' != ''"
-      Inputs="$(_ResInputFileFullPath)"
-      Outputs="$(_ResOutputFileFullPath);$(_ResTxtFullPath)">
-
-    <PropertyGroup>
-      <ResArgs Condition="'$(Configuration)' != 'Release'">$(ResArgs) /debug</ResArgs>
-      <ResArgs Condition="'$(ResInputPath)'  != ''">$(ResArgs) &quot;$(_ResInputFileFullPath)&quot;</ResArgs>
-      <ResArgs>$(ResArgs) &quot;$(_ResOutputFileFullPath)&quot;</ResArgs>
-      <ResArgs>$(ResArgs) $(AssemblyName)</ResArgs>
-
-      <ResCmd>&quot;$(_ResPath)&quot; $(ResArgs)</ResCmd>
-    </PropertyGroup>
-
-    <PropertyGroup>
-      <ResInputs>$(ResInputs) $(_ResInputFileFullPath)</ResInputs>
-    </PropertyGroup>
-
-    <Exec Command="$(ResCmd)" StandardOutputImportance="low" />
-    <Exec Command="$(_JSon2TxtPath) $(ResInputs) /out $(_ResTxtFullPath)" StandardOutputImportance="low" />
+      Condition="'$(StringResourcesPath)'!=''"
+      Inputs="$(StringResourcesPath)"
+      Outputs="$(IntermediateResOutputFileFullPath)">
+    
+    <GenerateResourcesCode 
+        ResxFilePath="$(StringResourcesPath)" 
+        IntermediateFilePath="$(IntermediateResOutputFileFullPath)" 
+        OutputSourceFilePath="$(ResourceSRPath)" 
+        AssemblyName="$(AssemblyName)" >
+    </GenerateResourcesCode>
 
     <ItemGroup>
-      <Compile Include="$(_ResOutputFileFullPath)" />
-      <FileWrites Include="$(_ResOutputFileFullPath)"/>
-      <FileWrites Include="$(_ResTxtFullPath)"/>
+      <FileWrites Include="$(IntermediateResOutputFileFullPath)" />
     </ItemGroup>
   </Target>
-
-  <Target Name="GenerateStringResources">
-    <ItemGroup>
-      <EmbeddedResource Include="$(_ResTxtFullPath)" Condition="Exists($(_ResTxtFullPath))">
-        <!-- "Resx" indicates it should go through GenerateResource -->
-        <Type>Resx</Type>
-        <WithCulture>false</WithCulture>
-        <ManifestResourceName>%(FileName)</ManifestResourceName>
-        <LogicalName>System.SR.resources</LogicalName>
-      </EmbeddedResource>
-    </ItemGroup>
-  </Target>
-
-</Project>
\ No newline at end of file
+  
+  <Import Project="ResourcesIncludes.targets" Condition="Exists('$(StringResourcesPath)') And Exists('$(ResourceSRPath)') And'$(ExcludeResourcesTarget)'==''"/>
```

Comment:
I'll call it SkipResourcesIncludesTarget as now the target we skip is called ResourcesIncludes